.travis.yml

```diff
@@ -17,44 +20,62 @@ cache:
     - $HOME/.composer/cache
     - $HOME/.config/composer/cache
 
+# Git clone depth.
+git:
+  depth: 1
+
 matrix:
   include:
-    - php: 5.4
-      env: WP_VERSION=4.9.10 PHP_LINT=1 ES_VERSION=1.3.0
     - php: 5.6
-      env: WP_VERSION=latest PHP_LINT=1 ES_VERSION=1.7.6
+      env: WP_VERSION=4.9.10 PHP_LINT=1 ES_VERSION=1.7.6
+      dist: trusty
     - php: 7.0
-      env: WP_VERSION=nightly WP_PHPCS=1 ES_VERSION=5.6.0
-    - php: 7.1
       env: WP_VERSION=latest PHP_LINT=1 ES_VERSION=2.4.6
+      dist: trusty
+    - php: 7.1
+      env: WP_VERSION=latest PHP_LINT=1 ES_VERSION=5.6.15
+      dist: xenial
     - php: 7.2
-      env: WP_VERSION=nightly PHP_LINT=1 ES_VERSION=6.4.1
+      env: WP_VERSION=latest PHP_LINT=1 ES_VERSION=6.8.0
+      dist: xenial
+    - php: 7.3
+      env: WP_VERSION=nightly PHP_LINT=1 WP_PHPCS=1 ES_VERSION=7.1.1
+      dist: xenial
 
 install:
   - bash bin/install-es.sh $ES_VERSION
 
+# Prepare your build for testing.
+# Failures in this section will result in build status 'errored'.
 before_script:
   - export PATH="$HOME/.config/composer/vendor/bin:$HOME/.composer/vendor/bin:$PATH"
-  - |
-    if [ -f ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini ]; then
-      phpenv config-rm xdebug.ini
-    else
-      echo "xdebug.ini does not exist"
-    fi
+
+  # Turn off Xdebug. See https://core.trac.wordpress.org/changeset/40138.
+  - phpenv config-rm xdebug.ini || echo "Xdebug not available"
+
   - |
     if [[ ! -z "$WP_VERSION" ]] ; then
       bash bin/install-wp-tests.sh wordpress_test root '' localhost $WP_VERSION
-      composer global require "phpunit/phpunit=4.8.*|5.7.*"
     fi
+
+  # Couple the PHPUnit version to the PHP version.
+  - |
+    case "$TRAVIS_PHP_VERSION" in
+    5.6)
+      echo "Using PHPUnit 4.8"
+      composer global require "phpunit/phpunit=4.8.*"
+      ;;
+    *)
+      echo "Using PHPUnit 6.1"
+      composer global require "phpunit/phpunit=6.1.*"
+      ;;
+    esac
+
   - |
     if [[ "$WP_PHPCS" == "1" ]]; then
       composer global require automattic/vipwpcs
       phpcs --config-set installed_paths $HOME/.composer/vendor/wp-coding-standards/wpcs,$HOME/.composer/vendor/automattic/vipwpcs
     fi
-  - sleep 5
-  - curl localhost:9200 && echo "ES is up" || (cat /tmp/elasticsearch.log && exit 1)
```

Comment:
The reason why this was not in the `install-es.sh` script was to give ES a chance to boot up in the background while travis did other things. This can shed a noticeable amount of time off the travis build.

Reply:
Yep, this is still a WIP PR, but I can move this logic to the end where it was before. Mostly I wanted to not `sleep` unless it was necessary, and to continue to `sleep` and wait on the load for longer if it *was* necessary, but there's no reason why we can't do that in the same location in the script where the logic was before. During initial testing I wanted to fail fast if it didn't come up, because that was the issue I was debugging at the time.